src/server/scene/mediating_display_changer.cpp

```diff
@@ -405,6 +449,13 @@ void ms::MediatingDisplayChanger::apply_config(
                 [this] { compositor->start(); }};
             display->configure(*conf);
         }
+        else if (configuration_changes_require_recompositing(*existing_configuration, *conf))
+        {
+            // Workaround for #2807: we could do better if Compositor had a `schedule_compositing(damage)` method
```

Comment:
Is there any particular reason *not* to introduce `Compositor::schedule_compositing` here? Tearing down and rebuilding the compositor is reasonably heavy, and if adding a method to `Compositor` isn't hard maybe we should do that instead?

Reply:
That would be an optimisation and, obviously, I considered it. But that opens a whole new rabbit hole that needs implementing across platforms. How about landing this and I'll raise an issue?

Reply:
:+1:, this will only run every so often, so shouldn't be a real performance issue?